Check atom balance

alongd opened the pull request:
fixes #369 

Check atom balance of Species and Reactions as they are defined by users

dranasinghe commented on arc/reactionTest.py:
```diff
@@ -119,6 +122,20 @@ def test_determine_multiplicity(self):
+    def test_check_atom_balance(self):
+        """Test the Reaction check_atom_balance method"""
+        rxn1 = ARCReaction(reactants=['CH4', 'OH'], products=['CH3', 'H2O'])
+        rxn1.r_species = [ARCSpecies(label='CH4', smiles='C'), ARCSpecies(label='OH', smiles='[OH]')]
```
I tested the following reaction "CH4 + OH + H2O <=> CH3 + H2O + H2O" technically it should be balance. Could you take a look at the modifications I did to unit tests?  
`        rxn1 = ARCReaction(reactants=['CH4', 'OH','H2O'], products=['CH3', 'H2O','H2O'])
        rxn1.r_species = [ARCSpecies(label='CH4', smiles='C'), ARCSpecies(label='OH', smiles='[OH]'), ARCSpecies(label='H2O', smiles='O')]
        rxn1.p_species = [ARCSpecies(label='CH3', smiles='[CH3]'), ARCSpecies(label='H2O', smiles='O'), ARCSpecies(label='H2O', smiles='O')]
        self.assertTrue(rxn1.check_atom_balance())`

alongd commented:
Thanks for looking at the PR, and for the suggestion!
I implemented your test suggestion of having the same species on one side of the reaction. The `count` parameter by which entries that are added to `r_well` and `p_well` are multiplied should address such cases.
I modified the test a bit, `products` indeed has two H2O species labels, but `p_species` points to the actual objects calculated, so it holds only one H2O species (these are the actual species for which QM jobs were spawned, in this example we will only need to calculate one H2O species, but we'll use its E0 twice for the reaction).
Let me know if you have any additional comments

alongd commented:
@dranasinghe, I added the bug fix for multiple occurrences of species in a reaction well to this PR since it requires features introduced here (counting species occurrences from the reaction label).
Please take a look.
Could you try restarting your ARC project on this branch and see if you get a reasonable rate now?
